### PR TITLE
[CHK-2319] -  checkout redirect implementation api client configuration

### PR DESCRIPTION
**Pull request by pietro-tota**
Depends on https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/396

#### List of Changes
Add Checkout Redirect api client configuration.
Since b2b Redirection api's will be implemented by every PSP this configuration handles PSP to api client mapping returning the api client configured for each PSP dinamically

#### Motivation and Context
This implementation handle api client for PSP dynamic generation

#### How Has This Been Tested?
Junit tests + tests with eCommerce local

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: api-spec/redirect/redirect-api.yaml [hunk 1]
```diff
@@ -0,0 +1,218 @@
+openapi: 3.0.0
```
Comment (infantesimone) [inline api-spec/redirect/redirect-api.yaml:1]: is this openapi used for client generation? could we insert a folder in the file path that indicates this? for example `client/openapi/*.yaml` or `openapi/client/*.yaml`

**Review by infantesimone — DISMISSED** [review]

File: api-spec/redirect/redirect-api.yaml (hunk 1 above)
Comment (pietro-tota) [reply api-spec/redirect/redirect-api.yaml:1]: Yes I like it. Can we leave api-spec as root folder or maybe is preferable to move it out?

Comment (pietro-tota) [reply api-spec/redirect/redirect-api.yaml]: Fixed with 99dd8836a5ea433dadf7c77a54c5de8b9a0475e6

File: api-spec/client/openapi/redirect/redirect-api.yaml [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      required: true
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/RedirectUrlRequest'
+  securitySchemes:
+    PspApiKeyAuth:
+      type: apiKey
+      in: header
+      name: x-api-key
```
Comment (infantesimone) [inline api-spec/client/openapi/redirect/redirect-api.yaml:217]: TO DO: add refs to "nuova connettività" (and update api header `ocp-apim-subscription-key`)

Comment (pietro-tota) [reply api-spec/client/openapi/redirect/redirect-api.yaml:217]: Ok, seems legit. Just one doubt, this way PSP implementation will expect api key to be present into ocp-apim-subscription-key header? Or this modification affers to api exposed by us to PSP (aka the outcome request)?

Comment (giovanniberti) [reply api-spec/client/openapi/redirect/redirect-api.yaml:217]: Shouldn't this header be only on the API interface exposed on `pagopa-api`?

Comment (pietro-tota) [reply api-spec/client/openapi/redirect/redirect-api.yaml:217]: @infantesimone merging this pr as is for the api key header name, will be updated in the future after integration clarification

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**